api/src/test/java/org/pmiops/workbench/institution/InstitutionEmailAddressMapperTest.java

```diff
@@ -0,0 +1,117 @@
+package org.pmiops.workbench.institution;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailAddress;
+import org.pmiops.workbench.model.Institution;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@Import(InstitutionEmailAddressMapperImpl.class)
+@DataJpaTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class InstitutionEmailAddressMapperTest {
+  @Autowired InstitutionEmailAddressMapper mapper;
+
+  @Test
+  public void test_modelToDb() {
+    // contains an out-of-order duplicate
+    final List<String> rawAddresses =
+        Lists.newArrayList("alice@nih.gov", "joel@other-inst.org", "alice@nih.gov");
+    final SortedSet<String> sortedDistinctAddresses = new TreeSet<>(rawAddresses);
+
+    final Institution modelInst =
+        new Institution()
+            .shortName("Broad")
+            .displayName("The Broad Institute")
+            .emailAddresses(rawAddresses);
+
+    // does not need to match the modelInst; it is simply attached to the DbInstitutionEmailAddress
+    final DbInstitution dbInst = new DbInstitution();
+
+    final Set<DbInstitutionEmailAddress> dbAddresses = mapper.modelToDb(modelInst, dbInst);
+
+    assertThat(dbAddresses.size()).isEqualTo(sortedDistinctAddresses.size());
+
+    for (final DbInstitutionEmailAddress dbAddress : dbAddresses) {
```

Comment:
nit: you can do this without a loop if you like by asserting with contains all and doing the stream operation for all match.